docs/source/distributed.algorithms.join.rst

```diff
@@ -0,0 +1,14 @@
+.. role:: hidden
+    :class: hidden-section
+
+Generic Join Context Manager
+============================
+
```

Comment:
Shall we add a short paragraph describing the purpose of this join context manager?

Reply:
TODO: when this lands, and when the tutorial lands, let's also add a link to this doc page to pointing to the tutorial page.